### DISCUSSION TITLE
Accessing variables from other scenes/AutoLoads/etc.

### DISCUSSION THREAD

**Discussion by Novolin** [dialogic-godot/dialogic#778] [Q&A] [answered]
I was hoping to use Dialogic to organize the story and general text box stuff for an RPG-style game I'm making, and I was wondering how doable it is to have it handle variables that are defined elsewhere in a scene. For example, I have a dictionary of objects with descriptions, formatted like this:

```
{
	"Dunkey Kong":{
		"description":"An Italian-American basketball player challenges a gorilla to a slam dunk contest.",
		"released":true,
		"owned":false,
		"slot": null
	}
}
```
and my current (very messy) implementation is as follows: 
```
func _on_Player_interact(obj):
	#code that sets what the title and content should be, usually by looking up in a dictionary
	var title = "This string is the title"
	var content = "this string is the content"
	emit_signal("openDialog", title, content)
```
(they are then shown through a rich text node)

and I wanted the text box to display the object name as the title, and the description as the content.

Is it possible to call a dialog box with just a string for the title and rich text? 

I'm planning on using they system to handle cutscenes, which it will work really well for without having to much hacky stuff.

**Comment by Jowan-Spooner** [comment, marked as answer]
If this (title/content) was all you want your textbox to display during the whole game, I would most definitely not recommend using dialogic. 
You can use dialogic to display text that is defined during the runtime, but as it is not the originally intended way of doing it, it is a bit clunky. There should be an [example in the FAQ](https://github.com/coppolaemilio/dialogic/blob/main/addons/dialogic/Documentation/Content/FAQ/create-timeline-using-gdscript.md). 
But again, only do it this way if you intend on using dialogic anyway for other purposes. The EXECUTING side of dialogic is it's weakness imo, while the editor is it's strength.

**Reply by Novolin** [reply]
I think I just missed that page lmao

It makes formatting cutscenes/text-heavy parts of it much easier, and the editor is fantastic for what I want to do with them, it was just one little piece i was struggling with since I wanted to keep all of my formatting pretty consistent
